sqswatcher/plugins/sge.py

```diff
@@ -95,33 +126,51 @@ def addHost(hostname, cluster_user, slots):
         time.sleep(1)
     ssh.close()
 
-    # Add the host to the qll.q
+    # Add the host to the all.q
     command = ('/opt/sge/bin/lx-amd64/qconf -aattr hostgroup hostlist %s @allhosts' % hostname)
-    __runSgeCommand(command)
+    __run_sge_command(command)
 
     # Set the numbers of slots for the host
     command = ('/opt/sge/bin/lx-amd64/qconf -aattr queue slots ["%s=%s"] all.q' % (hostname,slots))
-    __runSgeCommand(command)
+    __run_sge_command(command)
+
 
-def removeHost(hostname,cluster_user):
+def removeHost(hostname, cluster_user):
     log.info('Removing %s', hostname)
 
-    # Removing host as administrative host
-    command = ("/opt/sge/bin/lx-amd64/qconf -dh %s" % hostname)
-    __runSgeCommand(command)
+    # Check if host is administrative host
+    command = "/opt/sge/bin/lx-amd64/qconf -sh"
+    if __is_host_in_command_output(command, hostname):
+        # Removing host as administrative host
+        command = ("/opt/sge/bin/lx-amd64/qconf -dh %s" % hostname)
+        __run_sge_command(command, raise_exception=True)
+    else:
+        log.info('Host %s is not administrative host', hostname)
 
+    # Check if host is in all.q (qconf -sq all.q)
     # Purge hostname from all.q
     command = ("/opt/sge/bin/lx-amd64/qconf -purge queue '*' all.q@%s" % hostname)
-    __runSgeCommand(command)
+    __run_sge_command(command)
```

Comment:
Should we add the `__is_host_in_command_output` check for this command too?

Reply:
I didn't add for two reasons:
* this command and the other below are non blocking, meaning the return code is always 0 also when you try to remove an host not present
* the format of the output returned by these two query command is not the same as the others and require a different parsing strategy

Reply:
Ok, thank you